lib/repl.js

```diff
@@ -689,8 +689,33 @@ function intFilter(item) {
   return /^[A-Za-z_$]/.test(item);
 }
 
+const ARRAY_LENGTH_THRESHOLD = 1e6;
+
+function mayBeLargeObject(obj) {
+  if (Array.isArray(obj)) {
+    return obj.length > ARRAY_LENGTH_THRESHOLD ? ['length'] : null;
+  } else if (utilBinding.isTypedArray(obj)) {
+    return obj.length > ARRAY_LENGTH_THRESHOLD ? [] : null;
```

Comment:
TypedArrays' `length` is implemented as a getter on the prototype instead of an own property, as is the case with Arrays.